QMTest/TestSCons.py

```diff
@@ -792,6 +792,15 @@ def java_where_jar(self, version=None):
             where_jar = self.where_is('jar', ENV['PATH'])
         if not where_jar:
             self.skip_test("Could not find Java jar, skipping test(s).\n")
+        elif sys.platform == "darwin":
```

Comment:
Can you throw this logic into a function?  It's the same code copy/pasted 3 x, and likely to get fixed in just one or not all places at some point causing pain...